include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2263 @@
+typedef struct SDL_GpuSampler SDL_GpuSampler;
+typedef struct SDL_GpuShader SDL_GpuShader;
+typedef struct SDL_GpuComputePipeline SDL_GpuComputePipeline;
+typedef struct SDL_GpuGraphicsPipeline SDL_GpuGraphicsPipeline;
+typedef struct SDL_GpuCommandBuffer SDL_GpuCommandBuffer;
+typedef struct SDL_GpuRenderPass SDL_GpuRenderPass;
+typedef struct SDL_GpuComputePass SDL_GpuComputePass;
+typedef struct SDL_GpuCopyPass SDL_GpuCopyPass;
+typedef struct SDL_GpuFence SDL_GpuFence;
+
+typedef enum SDL_GpuPrimitiveType
+{
+    SDL_GPU_PRIMITIVETYPE_POINTLIST,
+    SDL_GPU_PRIMITIVETYPE_LINELIST,
+    SDL_GPU_PRIMITIVETYPE_LINESTRIP,
+    SDL_GPU_PRIMITIVETYPE_TRIANGLELIST,
+    SDL_GPU_PRIMITIVETYPE_TRIANGLESTRIP
+} SDL_GpuPrimitiveType;
+
+typedef enum SDL_GpuLoadOp
+{
+    SDL_GPU_LOADOP_LOAD,
+    SDL_GPU_LOADOP_CLEAR,
+    SDL_GPU_LOADOP_DONT_CARE
+} SDL_GpuLoadOp;
+
+typedef enum SDL_GpuStoreOp
+{
+    SDL_GPU_STOREOP_STORE,
+    SDL_GPU_STOREOP_DONT_CARE
+} SDL_GpuStoreOp;
+
+typedef enum SDL_GpuIndexElementSize
+{
+    SDL_GPU_INDEXELEMENTSIZE_16BIT,
+    SDL_GPU_INDEXELEMENTSIZE_32BIT
+} SDL_GpuIndexElementSize;
+
+typedef enum SDL_GpuTextureFormat
+{
+    SDL_GPU_TEXTUREFORMAT_INVALID = -1,
+
+    /* Unsigned Normalized Float Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8,
+    SDL_GPU_TEXTUREFORMAT_B8G8R8A8,
+    SDL_GPU_TEXTUREFORMAT_B5G6R5,
+    SDL_GPU_TEXTUREFORMAT_B5G5R5A1,
+    SDL_GPU_TEXTUREFORMAT_B4G4R4A4,
+    SDL_GPU_TEXTUREFORMAT_R10G10B10A2,
+    SDL_GPU_TEXTUREFORMAT_R16G16,
+    SDL_GPU_TEXTUREFORMAT_R16G16B16A16,
+    SDL_GPU_TEXTUREFORMAT_R8,
+    SDL_GPU_TEXTUREFORMAT_A8,
+    /* Compressed Unsigned Normalized Float Color Formats */
+    SDL_GPU_TEXTUREFORMAT_BC1,
+    SDL_GPU_TEXTUREFORMAT_BC2,
+    SDL_GPU_TEXTUREFORMAT_BC3,
+    SDL_GPU_TEXTUREFORMAT_BC7,
+    /* Signed Normalized Float Color Formats  */
+    SDL_GPU_TEXTUREFORMAT_R8G8_SNORM,
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8_SNORM,
+    /* Signed Float Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R16_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R16G16_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R16G16B16A16_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R32_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R32G32_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R32G32B32A32_SFLOAT,
+    /* Unsigned Integer Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R8_UINT,
+    SDL_GPU_TEXTUREFORMAT_R8G8_UINT,
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8_UINT,
+    SDL_GPU_TEXTUREFORMAT_R16_UINT,
+    SDL_GPU_TEXTUREFORMAT_R16G16_UINT,
+    SDL_GPU_TEXTUREFORMAT_R16G16B16A16_UINT,
+    /* SRGB Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8_SRGB,
+    SDL_GPU_TEXTUREFORMAT_B8G8R8A8_SRGB,
+    /* Compressed SRGB Color Formats */
+    SDL_GPU_TEXTUREFORMAT_BC3_SRGB,
+    SDL_GPU_TEXTUREFORMAT_BC7_SRGB,
+    /* Depth Formats */
+    SDL_GPU_TEXTUREFORMAT_D16_UNORM,
+    SDL_GPU_TEXTUREFORMAT_D24_UNORM,
+    SDL_GPU_TEXTUREFORMAT_D32_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_D24_UNORM_S8_UINT,
+    SDL_GPU_TEXTUREFORMAT_D32_SFLOAT_S8_UINT
```

Comment:
No STENCIL8 format that has no depth component?

Reply:
If it's supported everywhere, sure (pretty sure it is...), filed as https://github.com/thatcosmonaut/SDL/issues/98